### ISSUE TITLE
[QUESTION] Has anyone noticed WebRTC differences between MacOS Catalina and Big Sur?

### ISSUE THREAD

**Issue by matt-donahoe-skydio** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1026]
I have a signaling client built from the [kinesis js sdk example](https://awslabs.github.io/amazon-kinesis-video-streams-webrtc-sdk-js/examples/index.html), which connects to a master running with this c sdk, [release v1.3.0](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/releases/tag/v1.3.0).

Everything works great in Ubuntu and MacOS Catalina, but on MacOS Big Sur, I get this error in the Chrome developer console:
```
Uncaught (in promise) DOMException: Failed to execute 'setRemoteDescription' on 'RTCPeerConnection': Failed to set remote answer sdp: Failed to set remote video description send parameters for m-section with mid='0'.
```

Is there some new offer-answer protocol that the Kinesis c sdk doesn't handle? Has anyone else seen this?

**Comment by disa6302** [comment]
@matt-donahoe-skydio,

Hmm. Strange. Offer and answer exchange shouldn't be platform dependent. Can you explain any changes you made to the JS and C SDK to get it working on MacOS Catalina and Ubuntu? What I am looking for is specifically any changes made to transceiver direction and transceiver selection on JS and C SDK.

Also, can you attach the logs and the SDPs (run `export DEBUG_LOG_SDP=TRUE`) for the two MacOS platforms?

**Comment by matt-donahoe-skydio** [comment]
Other than a massive refactor of all the code, I have two relevant changes: 

1. My client isn't sending video or audio streams anymore, only a data channel. It doesn't require a webcam to be a client.
2. My master does not create an audio stream, only video. I commented out the `addTransceiver` line for the audioTrack:  https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/master/samples/Common.c#L463

I'll see if I can reproduce the behavior with a minimal example.

**Comment by MushMal** [comment]
Quick question. You are using the embedded C WebRTC SDK on MacOS right? If so, there shouldn't be any issues with the versions. There had been some issues with Catalina as it has system protection but nothing so far that points on any issues with big sur

**Comment by matt-donahoe-skydio** [comment]
No, the embedded sdk is running on my Ubuntu xenial machine as the master, and my javascript client is running on MacOS Big Sur

**Comment by matt-donahoe-skydio** [comment]
Just discovered that other browsers have errors on *both* Catalina and Big Sur, so perhaps Chrome 87.0 on Catalina does something different than all the rest, including the same Chrome 87.0 on Big Sur.

Latest Firefox on catalina and big sur fail with the same error
```
 Uncaught (in promise) DOMException: Answer tried to set recv when offer did not set send
```

Latest Safari on catalina and big sur has this error
```
Unhandled Promise Rejection: InvalidAccessError: Failed to set remote answer sdp: The order of m-lines in answer doesn't match order in offer. Rejecting answer.
```

**Comment by MushMal** [comment]
OK, sorry for the confusion. The issue is filed in the embedded C WebRTC SDK and I was not sure what the deal is. On non-WebRTC clients we don't own the implementation of the WebRTC. In case of JS, it's the browsers implementation which is likely be based on Googles code. Please resolve this issue and cut a new issue in the right repository. It's likely to be something with an access issue.

**Comment by MushMal** [comment]
One more thing... There have been some fixes/changes that went in for embedded WebRTC sdk that deal with the order/m-lines in the SDP.

**Comment by matt-donahoe-skydio** [comment]
My suspicion is that I'm not properly constructing the correct answer, so I felt it was relevant to the c sdk, and I was hoping for pointers on what I should do. I believe I filed this as a QUESTION not a BUG.

**Comment by matt-donahoe-skydio** [comment]
I found this
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/966

@MushMal  Are there other changes I should be sure that I have? Do you know if it m-lines changes you're thinking of are in the 1.4 release, or should I just be running master?

**Comment by MushMal** [comment]
Sure, one more question. How are you constructing your answer? The SDK handles that. Try to run the stock samples first. Make sure everything works as expected in your setup. Then, try to modify and add your custom changes.

Just saw your next message. The master branch is our "stable" branch. All of the tests are passing on it at all times. We are cutting releases from time-to-time based on a set of features that we package. The release process assumes extra scrutiny in checking the release candidate and running our canaries from demo repository on it.

Generally, my recommendation would be to use the tip of the master branch for your development and stabilization of your application. Once this is done, you can then rebase it on a next upcoming release.

**Comment by matt-donahoe-skydio** [comment]
Ok cool. I'll look at those m-lines changes and create a new issue when I have a clearer idea of what I'm asking.

Thanks!
